tests/unit/lib/utils/test_tar.py

```diff
@@ -19,7 +19,42 @@ def test_generating_tarball(self, temporary_file_patch, tarfile_open_patch):
 
         tarfile_file_mock.add.assert_called()
         tarfile_file_mock.add.assert_has_calls(
-            [call("/some/path", arcname="/layer1"), call("/some/dockerfile/path", arcname="/Dockerfile")],
+            [
+                call("/some/path", arcname="/layer1", filter=None),
+                call("/some/dockerfile/path", arcname="/Dockerfile", filter=None),
+            ],
+            any_order=True,
+        )
+
+        temp_file_mock.flush.assert_called_once()
+        temp_file_mock.seek.assert_called_once_with(0)
+        temp_file_mock.close.assert_called_once()
+        tarfile_open_patch.assert_called_once_with(fileobj=temp_file_mock, mode="w")
+
+    @patch("samcli.lib.utils.tar.tarfile.open")
+    @patch("samcli.lib.utils.tar.TemporaryFile")
+    def test_generating_tarball_with_filter(self, temporary_file_patch, tarfile_open_patch):
+        temp_file_mock = Mock()
+        temporary_file_patch.return_value = temp_file_mock
+
+        tarfile_file_mock = Mock()
+        tarfile_open_patch.return_value.__enter__.return_value = tarfile_file_mock
+
+        def tar_filter(tar_info):
+            tar_info.mode = 0o500
```

Comment:
Can we add an assert that verifies the `tar_info.mode` for each file is set correctly?

Reply:
tarfile_file_mock.add.assert_has_calls(..., filter=tar_filter) asserts that the filter must be used as parameter for tarfile library.
It will not be actually called in the unittest.